src/listener.coffee

```diff
@@ -39,7 +39,7 @@ class TextListener extends Listener
   # callback - A Function that is triggered if the incoming message matches.
   constructor: (@robot, @regex, @callback) ->
     @matcher = (message) =>
-      if message instanceof TextMessage
+      if message.match
```

Comment:
Nope, this won't work. Not all messages with a `match` method have a `@regex` property. You need to do `if message.type == 'text'`